src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindArrayStatistics.cpp

```diff
@@ -467,6 +499,29 @@ const std::atomic_bool& FindArrayStatistics::getCancel()
   return m_ShouldCancel;
 }
 
+// -----------------------------------------------------------------------------
+void FindArrayStatistics::sendMessage(const std::string& action)
+{
+  m_MessageHandler(IFilter::ProgressMessage{IFilter::Message::Type::Info, fmt::format("Now Processing: {}", action)});
+}
+
+// -----------------------------------------------------------------------------
+void FindArrayStatistics::sendThreadSafeProgressMessage(size_t counter)
+{
+  std::lock_guard<std::mutex> guard(m_ProgressMessage_Mutex);
+
+  m_ProgressCounter += counter;
+
+  auto now = std::chrono::steady_clock::now();
+  if(std::chrono::duration_cast<std::chrono::milliseconds>(now - m_InitialTime).count() > 1000) // every second update
+  {
+    auto progressInt = static_cast<size_t>((static_cast<double>(m_ProgressCounter) / static_cast<double>(m_TotalElements)) * 100.0);
```

Comment:
@nyoungbq I'm probably just missing this in the code but just in case I'm not where are we setting the value of m_TotalElements? The header file initializes it to 0 but I don't see where we set this to a usable value.